modules/dna_rna_tools.py

```diff
@@ -0,0 +1,273 @@
+from . import bases
+from typing import Tuple, Union, TypedDict
+
+
+class Options(TypedDict):
+    '''
+    Default structure of dna_rna_tools options.
+
+    seq_type (str): Sequence type of input nucleic acids.
+        'rna' and 'dna' are accepted by downstream functions.
+        None value triggers get_data function to check nucleic
+        acid type in input.
+    frame (int): Number of reading frame to translate. Accepted
+        values are 1, 2, 3.
+    '''
+    seq_type: Union[None, str]
+    frame: int
+
+
+def get_data(*seqs_and_action: str,
+             **options: Options
+             ) -> Tuple[Tuple[str], str, Options]:
+    '''
+    Orginizes data to fit run_dna_rna_tools function structure.
+
+    Splits positional arguments into sequences and action, validates
+    sequences and forms a complete class Options dictionary.
+
+    Arguments
+    ---------
+    *seqs_and_action (str): any number of nucleic, followed by action.
+
+    **options: Additional keyword arguments.
+    - seq_type (str): Type of nucleic acid to be processed by
+    rna_dna_tools.
+    - frame (int): Specifies the reading frame
+    for the 'translate' action.
+    Returns
+    -------
+
+    Tuple[Tuple[str], str, Options]: structured data which can be processed
+    by run_rna_dna_tools.
+    '''
+    action = str(seqs_and_action[-1])
+    options = add_defaults(**options)
+    seqs = seqs_and_action[:-1]
+    if is_nucleic_acid(seqs) is False:
+        raise ValueError("Not a nucleic acid")
+    if options["seq_type"] is None:
+        options["seq_type"] = is_dna_or_rna(seqs)
+
+    return seqs, action, options
+
+
+def add_defaults(**options: Options) -> Options:
+    '''
+    Add default values to class Options dictionary if some
+    are not specified.
+
+    Arguments
+    ---------
+    **options: Additional keyword arguments.
+
+    - seq_type (str): a sequence type specified.
+    - frame (int): a reading frame specified.
+
+    Returns
+    -------
+    dict (Options): A dictionary class Options with user specified
+    values and defualt values if one wasn't provided. Default values:
+    - seq_type: None
+    - frame: 1
+    '''
+    default_options = {"seq_type": None, "frame": 1}
+    default_options.update(**options)
+    return default_options
+
+
+def is_nucleic_acid(seqs: tuple) -> bool:
+    '''
+    Checks if all of sequences passed are valid nucleic acids
+
+    Valid nucleic acids contain only {A, T, G, C, U} bases,
+    do not contain T and U simultaneously, must not be
+    blank strings.
+
+    Arguments
+    ---------
+    seqs (Tuple[str]): one or more sequences.
+
+    Returns
+    --------
+    bool: True if condtions are satisfied for all sequences.
+    False otherwise.
+    '''
+    seqs = tuple(seq.upper() for seq in seqs)
+    return all(seq and set(seq).issubset(bases.nucleotides) for seq in seqs)
```

Comment:
Ну тогда уж... А то мало ли тебе дадут список, скажут `is_nucleic_acid`, и ты ответишь `True`. А это ведь не нукл.кислота, это список нукл.кислот.
```suggestion
def is_nucleic_acids(seqs: tuple) -> bool:
    '''
    Checks if all of sequences passed are valid nucleic acids

    Valid nucleic acids contain only {A, T, G, C, U} bases,
    do not contain T and U simultaneously, must not be
    blank strings.

    Arguments
    ---------
    seqs (Tuple[str]): one or more sequences.

    Returns
    --------
    bool: True if condtions are satisfied for all sequences.
    False otherwise.
    '''
    seqs = tuple(seq.upper() for seq in seqs)
    return all(seq and set(seq).issubset(bases.nucleotides) for seq in seqs)
```

Reply:
Да, коллекции во множественном числе, иногда упускаю. Буду следить